Module_4A/countries_data_handling.rb

```diff
@@ -0,0 +1,129 @@
+# ---------------- military module ---------------- 
+module MilitaryModule
+    attr_accessor :army_strength
+  
+    def initArmyStrength(army_strength)
+      @army_strength = army_strength
+    end
+  
+    def war_win_probability
+      return @army_strength > 300000
+    end
+  end
+  
+  
+  # ------------------------ loan_module ------------------------
+  module LoanModule
+    def loan_eligibility
+      # to determine whether the country is eligible for a loan from IMF or WB
+      return @gdp > 1000000
+    end
+  end
+  
+  # ----------------------- security_council_module -------------------------
+  module SecurityCouncilModule
+    def un_security_council_seat_eligibility
+      # to determine whether the country is eligible for a seat in the UN Security Council
+      return @army_strength > 500000
+    end
+  end
+  
+  
+  # ------------------------ country Class -----------------------------------
+  class Country
+    include LoanModule
+    include SecurityCouncilModule
+    include MilitaryModule
+  
+    attr_accessor :name, :population, :gdp, :states, :devStatus
+  
+    def initialize(name, population, gdp, states, army_strength, devStatus)
+      @name = name
+      @population = population
+      @gdp = gdp
+      @states = states
+      initArmyStrength(army_strength)
+      @devStatus = devStatus
+    end
+  end
+  
+
+#  ----------------- country data class---------------------------
+  class CountryData
+    def initialize
+      @countries = []
+    end
+  
+    def addCountry(c)
+      @countries << c
+    end
+  
+    def dispCountry
+      @countries.each do |c|
+        puts "#{c.name} - Population: #{c.population}, GDP: #{c.gdp}, States: #{c.states}, Army Strength: #{c.army_strength}, Development Status: #{c.devStatus}"
```

Comment:
Split such long lines into multiple lines for better readability

Reply:
Improved readability using puts and print